model_analyzer/config/run/run_config_generator.py

```diff
@@ -113,6 +113,8 @@ def generate_run_config_for_model_sweep(self, model, model_sweep):
 
             model_name_index = self._model_name_index
             model_config_dict = model_config.get_config()
+            self._apply_model_config_defaults(model_config_dict,
```

Comment:
>For example, an unspecified instance group was considered "different" than an instance group with count=1 kind=GPU, even though they are the same. This change fixes that problem. However, the onus shouldn't be on this class to update the default. I suppose this functionality should go into ModelConfig class?

I think depending on the backend it could be a different configuration. My suggestion is to not try to go very deep on detecting whether two configurations are really equal to each other or not. We should just try to make sure that we have always run the config provided by the user. Two configurations are only equal when they look exactly like each other and are specifying the same fields.

Reply:
Reverted